### PR TITLE
added health bar

### PR DESCRIPTION
**Pull request by DhyaneshGhaghada**
## Summary

This PR implements health bar as described in issue #46 .
Structural Changes:
Created a new directory under images called 'health_bar' and new 'health_bar.py' script under 'src/gui'.

## Checklist
- [x] I have tested this change locally and it works as expected.
- [x] I have made sure that the code follows the formatting and style guidelines of the project.
- [x] Since this PR modifies related documentation, I have proofread files for spelling and grammar errors.

## Labels
`type: feature`, `game-ui`

### CODE REVIEW

File: src/support.py [hunk 1]
```diff
@@ -175,4 +175,12 @@ def tile_to_screen(pos):
...
\ No newline at end of file
+    return pos[0] // tile_size, pos[1] // tile_size
+
+def interpolate_color(color1, color2, factor):
+    ''' interpolate / blend two colors '''
+    return (
+        int(max(0, min(255, color1[0] + (color2[0] - color1[0]) * factor))),
+        int(max(0, min(255, color1[1] + (color2[1] - color1[1]) * factor))),
+        int(max(0, min(255, color1[2] + (color2[2] - color1[2]) * factor))),
+    )
```
Comment (fortwoone) [inline src/support.py:186]: You did not need to do that, as `pygame.Color` already implements a method for that, read the doc for it here: 
https://pyga.me/docs/ref/color.html#pygame.Color.lerp

**Review by fortwoone — CHANGES_REQUESTED** [review]
Just a slight nitpick

File: src/support.py (hunk 1 above)
Comment (DhyaneshGhaghada) [reply src/support.py]: Hello I have added pygame.Color.lerp instead of that custom function, any other suggestion?

**Review by fortwoone — APPROVED** [review]
LGTM

**Pull request merged**